src/brackets/index.js

```diff
@@ -4,6 +4,40 @@
  * @param {string} str The string of brackets.
  * @returns {"valid" | "invalid"} Whether or not the string is valid.
  */
-function isValid(str) {}
+
+
+function isValid(str) {
+let leftBracketArr = []; //will contain all left brackets
+let rightBracketArr = []; //will contain all right brackets
+for(let i=0; i < str.length; i++){
+  let singleBracket = str[i]; //isolate single char in input string
+  if(singleBracket === '(' || singleBracket === '[' || singleBracket === '{'){
+    leftBracketArr.push(str[i]);
+  }
+  let arrLength = leftBracketArr.length - 1; //gets the first index of half part in a recursive manner
```

Comment:
The comment here is incorrect. Which is why comments aren't very helpful sometimes. The variable name could also be better since that's not what it stores.

Reply:
okay, noted